gateway/errors.go

```diff
@@ -0,0 +1,18 @@
+package gateway
+
+import (
+	"fmt"
+)
+
+// Error represents the error returned by a handler to the requestor
+type Error struct {
+	Type    string `json:"type"`
+	Code    string `json:"code"`
```

Comment:
Should this be an int?

Reply:
It looks like this is the type? 🤔

Reply:
This is a copy of `marketplace.Error` where these are all strings. They need to be of the same basic type otherwise the `json.Decoder` will fail.